src/objects/aff/zcl_abapgit_json_path.clas.testclasses.abap

```diff
@@ -139,4 +142,82 @@ CLASS ltcl_json_path IMPLEMENTATION.
 
   ENDMETHOD.
 
+  METHOD deserialize_nested_arrays.
+    DATA: lt_file     TYPE string_table,
+          lo_cut      TYPE REF TO zcl_abapgit_json_path,
+          lv_xact     TYPE xstring,
+          lv_act      TYPE string,
+          lv_exp      TYPE string,
+          lt_exp      TYPE string_table,
+          lv_is_equal TYPE abap_bool.
+
+    APPEND `$.header.description=Text` TO lt_file.
+    APPEND `$.descriptions.methods[?(@.name=='METH1')].description=Sonne` TO lt_file.
+    APPEND `$.descriptions.methods[?(@.name=='METH1')].parameters[?(@.name=='param2')].description=ABC` TO lt_file.
+
+    CREATE OBJECT lo_cut.
+    lv_xact = lo_cut->deserialize( lt_file ).
+
+    APPEND `{  "header": { "description": "Text" } ,` TO lt_exp.
+    APPEND `"descriptions": {` TO lt_exp.
+    APPEND `"methods": [ ` TO lt_exp.
+    APPEND ` { "name": "METH1",` TO lt_exp.
+    APPEND `"description": "Sonne",` TO lt_exp.
+    APPEND `"parameters": [ { "name": "param2", "description": "ABC" } ]` TO lt_exp.
+    APPEND `}]}}` TO lt_exp.
+
+
+    lv_exp = concat_lines_of( table = lt_exp
+                              sep   = cl_abap_char_utilities=>newline ).
+
+    lv_act = zcl_abapgit_convert=>xstring_to_string_utf8( lv_xact ).
+
+    lv_is_equal = zcl_abapgit_ajson_utilities=>new( )->is_equal( iv_json_a = lv_act
+                                                                 iv_json_b = lv_exp ).
+
+    cl_abap_unit_assert=>assert_true( lv_is_equal ).
```

Comment:
I would prefer the usage of `assert_equal` so that we directly can see what is the difference between expected and actual value in the test results. 

However, I guess, this will increase efforts dramatically. Therefore, I guess it is fine like it is. Any thoughts?

Reply:
just added the assert_equal statement.

Reply:
fyi, assert_true/false don't exist in 702